core/src/main/java/nl/nn/adapterframework/jta/narayana/NarayanaConnectionFactoryFactory.java

```diff
@@ -20,25 +20,60 @@
 
 import org.jboss.narayana.jta.jms.ConnectionFactoryProxy;
 import org.jboss.narayana.jta.jms.JmsXAResourceRecoveryHelper;
+import org.jboss.narayana.jta.jms.TransactionHelper;
+import org.messaginghub.pooled.jms.JmsPoolXAConnectionFactory;
 
 import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
 
+import lombok.Getter;
 import lombok.Setter;
 import nl.nn.adapterframework.jndi.JndiConnectionFactoryFactory;
+import nl.nn.adapterframework.util.AppConstants;
 
 public class NarayanaConnectionFactoryFactory extends JndiConnectionFactoryFactory {
 
 	private @Setter NarayanaJtaTransactionManager transactionManager;
 
+	private @Getter @Setter int maxIdleTime = AppConstants.getInstance().getInt("transactionmanager.narayana.jms.connection.maxIdleTime", 60);
+	private @Getter @Setter int maxConnectionPoolSize = AppConstants.getInstance().getInt("transactionmanager.narayana.jms.connection.maxPoolSize", 20);
+	private @Getter @Setter int connectionCheckInterval = AppConstants.getInstance().getInt("transactionmanager.narayana.jms.connection.checkInterval", -1);
+	private @Getter @Setter int maxSessionPoolSize = AppConstants.getInstance().getInt("transactionmanager.narayana.jms.session.maxPoolSize", 20);
+
 	@Override
 	protected ConnectionFactory augment(ConnectionFactory connectionFactory, String connectionFactoryName) {
 		if (connectionFactory instanceof XAConnectionFactory) {
 			XAResourceRecoveryHelper recoveryHelper = new JmsXAResourceRecoveryHelper((XAConnectionFactory) connectionFactory);
 			this.transactionManager.registerXAResourceRecoveryHelper(recoveryHelper);
-			return new ConnectionFactoryProxy((XAConnectionFactory) connectionFactory, transactionManager.getTransactionHelper());
+
+			if(maxConnectionPoolSize > 1) {
+				return createConnectionFactoryPool(connectionFactory);
```

Comment:
Is the custom`NarayanaTransactionHelper` not needed for proper release of sessions / connections in this case?

Reply:
I mentioned this in the description of this PR, but simply put, no. In Spring Boot it used to be like that, until they removed it for some unknown reason.